File: SqlOrganize/Sql/Validation.cs
namespace SqlOrganize.Sql
{
    /*
    Validacion basica
    */
    public class Validation
    {

        public object? value { get; set; } //valor a validar;

        public List<(string msg, string? type)> errors { get; set; } = new(); //log de errores

        public Validation(object? _value) {
            value = _value;
        }

        public Validation Required()
        {
            if (value.IsNoE()) {
                errors.Add(("Valor nulo o vacío", "required"));
            }
            return this;
        }

        public Validation Type(string type)
        {
            switch (type)
            {
                case "string":
                    if(!value.IsNoE() && value is not String)
                        errors.Add(("Valor no texto", "type"));
                break;

                case "integer":
                case "int":
                    if (!value.IsNoE() && value is not int)
                        errors.Add(("Valor no entero", "type"));

                break;
            }
            return this;
        }

        public bool HasErrors() {
            return (errors.IsNoE()) ? false : true;
        }

        public Validation Clear()
        {
            errors.Clear();
            return this;
        }

    }
}

File: SqlOrganize/Sql/Validation.php
<?php

namespace SqlOrganize\Sql;

/**
 * Validacion basica
 */
class Validation
{
    public $value; // valor a validar
    public array $errors = []; // log de errores

    public function __construct($value)
    {
        $this->value = $value;
    }

    public function required(): self
    {
        if (empty($this->value)) {
            $this->errors[] = ["msg" => "Valor nulo o vacío", "type" => "required"];
        }
        return $this;
    }

    public function type(string $type): self
    {
        switch ($type) {
            case "string":
                if (!empty($this->value) && !is_string($this->value)) {
                    $this->errors[] = ["msg" => "Valor no texto", "type" => "type"];
                }
                break;

            case "integer":
            case "int":
                if (!empty($this->value) && !is_int($this->value)) {
                    $this->errors[] = ["msg" => "Valor no entero", "type" => "type"];
                }
                break;
        }
        return $this;
    }

    public function hasErrors(): bool
    {
        return !empty($this->errors);
    }

    public function clear(): self
    {
        $this->errors = [];
        return $this;
    }

}

?>